enterprise_subsidy/apps/api_client/tests/test_enterprise_catalog.py

```diff
@@ -83,14 +83,27 @@ def setUpTestData(cls):
         }
 
     @mock.patch('enterprise_subsidy.apps.api_client.base_oauth.OAuthAPIClient', return_value=mock.MagicMock())
-    def test_successful_fetch_course_content_metadata(self, mock_oauth_client):
+    def test_successful_fetch_course_content_metadata_for_customer(self, mock_oauth_client):
         """
-        Test the enterprise catalog client's ability to handle api requests to fetch content metadata from the catalog
-        service.
+        Test the enterprise catalog client's ability to handle api requests to fetch customer specific content metadata
+        from the catalog service.
         """
         mock_oauth_client.return_value.get.return_value = MockResponse(self.course_metadata, 200)
         enterprise_catalog_client = EnterpriseCatalogApiClient()
         response = enterprise_catalog_client.get_content_metadata_for_customer(
             self.enterprise_customer_uuid, self.course_key
         )
         assert response == self.course_metadata
+
+    @mock.patch('enterprise_subsidy.apps.api_client.base_oauth.OAuthAPIClient', return_value=mock.MagicMock())
+    def test_successful_fetch_course_content_metadata(self, mock_oauth_client):
+        """
+        Test the enterprise catalog client's ability to handle api requests to fetch content metadata from the catalog
+        service.
+        """
+        mock_oauth_client.return_value.get.return_value = MockResponse(self.course_metadata, 200)
+        enterprise_catalog_client = EnterpriseCatalogApiClient()
+        response = enterprise_catalog_client.get_content_metadata(self.course_key)
+        assert response == self.course_metadata
+        assert mock_oauth_client.return_value.get.call_args.args[0] == enterprise_catalog_client.metadata_endpoint
+        assert mock_oauth_client.return_value.get.call_args.kwargs == {'params': {'content_keys': ['edX+DemoX']}}
```

Comment:
One good thing to do here is to call `response = enterprise_catalog_client.get_content_metadata(self.course_key)` a second time to exercise the cache, and assert that the oauth client is still only called once.

Reply:
not a bad idea

Reply:
[inform] the caching happens at the metadata api level, so subsequent calls to the catalog api client should make multiple calls - I updated the content_metadata api caching test to ensure there's only one enterprise catalog client `get_content_metadata' call

Reply:
https://github.com/openedx/enterprise-subsidy/pull/203/files#diff-2c4756b0ae94b7023c130ba2f022f68ab167b4d1400155988f3ab24b3de3c91dR277